src/uint256.h

```diff
@@ -111,6 +112,47 @@ class uint256 : public base_blob<256> {
     static const uint256 ONE;
 };
 
+/* uint256 from const char *.
+ * Could theoretically have been a constructor but is overload of uint256S() for
+ * historical reasons.
+ */
+consteval uint256 uint256S(const char *str)
```

Comment:
Thanks for taking a closer look!

I want to enforce `consteval` for existing string literals. Ideally the `string_view` overload should be `constexpr` but from my research on MSVC assembly output in the PR summary it seems to handle lookup tables poorly. One possibility would be to remove the lookup table and use `if (c >= '0' && c <= '9')`, sacrificing some performance, but we don't often parse ASCII hex strings in runtime anyway, right?

The main purpose of overloading `uint256S()` with a `consteval` version was to avoid changing all the call-sites. If we are okay with changing call-sites, it might be better to introduce a `uint256(uint64_t,uint64_t,uint64_t,uint64_t)` constructor. That way it would be the compiler parsing hexadecimal integer literals directly. See my raw array (54e0213c9c7a7a942dd320264c1f9224c494b777) and span (bfc2fb049f2a12e9936b938fb073738babac0cbb) explorations.

Reply:
> > I want to enforce `consteval` for existing string literals. Ideally the `string_view` overload should be `constexpr` but from my research on MSVC assembly output
> 
> I think `string_view` can be used in a consteval context, no?

Yes, it's the rest of that paragraph that poses some concern.

Reply:
> Ideally the string_view overload should be constexpr but from my research on MSVC assembly output

Please don't be overly concerned about MSVC. We've had issues with it in the past, where it's failed to optimize code properly, or we've had to write workarounds for it, when code was otherwise fine in Clang or GCC, i.e (https://github.com/bitcoin/bitcoin/pull/28657#discussion_r1360780446). We don't use it as a release compiler (and never will), so if it's failing to do X, that isn't necessarily a blocker to making changes (if X is working correctly in Clang & GCC). MSVC can always be improved later, and shouldn't prevent us from writing better code now.

Reply:
As of 10baae11ecc376f2250d4a51d2bbfeea56c0a31d I have stopped overloading `uint256S()`:

Changed from `consteval uint256 uint256S(const char *str)` -> `consteval explicit uint256(const char (&str)[65])` and `base_blob(std::string_view str)`
This means the string width is enforced at compile time. (The new parsing code is much more strict and also asserts on the length).

Applied uint256S -> uint256 conversion where applicable, removing "0x"-prefixes.
TODO: if there is agreement on the current approach - create scripted diff commit.

Added `TxidFromStringS()` to mirror `uint256S()` for runtime use.

Clarified reason for introduction of `FixedVec` in commit message 10baae11ecc376f2250d4a51d2bbfeea56c0a31d.